geoalchemy2/__init__.py

```diff
@@ -103,19 +102,32 @@ def dispatch(event, table, bind):
                         stmt = stmt.execution_options(autocommit=True)
                         bind.execute(stmt)
                     elif bind.dialect.name == 'postgresql':
+
+                        index_name = "idx_{}_{}".format(table.name, c.name)
+
+                        if c.type.use_N_D_index:
```

Comment:
I never used ND index but does it make sense to use it on 2D data? Shouldn't we check that `dimension >= 3` or something like that?

Reply:
hmm, that is a good idea, if there is no higher dimensional data there should be no reason to index in higher dimensions!
Indeed from the [PostGIS documentation about N-D indexes ](https://postgis.net/workshops/postgis-intro/3d.html#n-d-indexes) they mention the following:

> For a higher dimension index to be useful, the data must cover a wide range of that dimension, relative to the kinds of queries you are constructing.

Reply:
Is it possible to create such index, even though it seems useless? In geoalchemy2 we don't want to add more constraints than postgis, so if it is possible to create ND index for 2d data we can keep it like this.

Reply:
I can't think of a use case, but indeed there is no constraint from PostGIS as far as I know

Reply:
Ok, let's not add new constraint then.